src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -994,78 +863,84 @@ private TypeGenerationSpec CreateTypeGenerationSpec(Type type, JsonSourceGenerat
 
                                 for (int i = 0; i < paramCount; i++)
                                 {
-                                    ParameterInfo parameterInfo = parameters![i];
-                                    TypeRef parameterTypeRef = EnqueueType(parameterInfo.ParameterType, generationMode);
+                                    IParameterSymbol parameterInfo = parameters![i];
+                                    TypeRef parameterTypeRef = EnqueueType(parameterInfo.Type, generationMode);
 
-                                    paramGenSpecArray[i] = new ParameterGenerationSpec()
+                                    paramGenSpecArray[i] = new ParameterGenerationSpec
                                     {
                                         ParameterType = parameterTypeRef,
                                         Name = parameterInfo.Name!,
-                                        HasDefaultValue = parameterInfo.HasDefaultValue,
-                                        DefaultValue = parameterInfo.GetDefaultValue(),
+                                        HasDefaultValue = parameterInfo.HasExplicitDefaultValue,
+                                        DefaultValue = parameterInfo.HasExplicitDefaultValue ? parameterInfo.ExplicitDefaultValue : null,
```

Comment:
Are you asking because the default is of type `object`? Because default method parameters can only be literals they should still behave fine w.r.t. caching. Besides, formatting should be the responsibility of the emitter in general.

Reply:
👍 for both points.